tenantid/tenantID.go

```diff
@@ -0,0 +1,25 @@
+package tenantid
+
+import (
+	"net/http"
+)
+
+const (
+	grpcMetadataHeaderKey = "grpc-metadata-archivist-internal-tenant-id"
+	headerKey             = "archivist-internal-tenant-id"
+	Prefix                = "tenant/"
+)
+
+func GetTenantIDFromHeader(h http.Header) string {
```

Comment:
How does this correspond to the functions in wal, particularly tenantid.go and middlware.go.

I think there is also some tenantID logic from headers in idtoken?

Reply:
yes - eventually most will be moved out of wal package and into datatrails-common but not yet.
This PR is the minumum required to move metrics.go from wal to datatrails-common

Reply:
Can we unit test these then?

Reply:
I would rather do that in separate PR/ticket...

Reply:
ticket raised for doing unittests on this repo

Reply:
perfect!